src/ck-core/ckfutures.C

```diff
@@ -426,6 +426,23 @@ public:
   }
 };
 
+// TODO :
+// a poorly located helper function for getting a buffer from a message
+// should be moved elsewhere (when PR#3228 is merged), and probably named ck::get_message_buffer(void*)
+void* CkGetMsgBuffer(void* msg) {
+  auto env = UsrToEnv(msg);
+  auto idx = env->getMsgIdx();
+  if (idx == CMessage_CkMarshallMsg::__idx) {
```

Comment:
Perhaps better as a `switch` statement?

Reply:
Hmm, the `__idx` members are not compile-time constants so it does not seem like they can be used as the value for a `case` statement (`error: expression must have a constant value`).

Reply:
Ah, I see.